docs/proposals/service-discovery.md

```diff
@@ -0,0 +1,98 @@
+
+`Service Scheme` - Services can be deployed using different schemes. Some popular schemes include `http`,`https`,`file`,`ftp` and `jdbc`.
+
+`Service Protocol` - Services use different protocols that clients need to speak in order to communicate with the service, some examples of service level protocols are SOAP, REST (Yes, technically REST isn’t a protocol but an architectural style). For service consumers it can be hard to tell what protocol is expected.
+
+## Service Description
+
+The API of a service is the point of interaction with a service consumer. The description of the API is an essential piece of information at creation time of the service consumer. It has become common to publish a service definition document on a know location on the service itself. This 'well known' place it not very standard, so it is proposed the service developer provides the service description path and the type of Definition Language (DL) used.
+
+`Service Description Path` - To facilitate the consumption of the service by client, the location this document would be greatly helpful to the service consumer. In some cases the client side code can be generated from such a document. It is assumed that the service description document is published somewhere on the service endpoint itself.
+
+`Service Description Language` - A number of Definition Languages (DL) have been developed to describe the service. Some of examples are `WSDL`, `WADL` and `Swagger`. In order to consume a decription document it is good to know the type of DL used.
+
+## Standard Service Annotations
+
+Kubernetes allows the creation of Service Annotations. Here we propose the use of the following standard annotations
+
+* `api.kubernetes.io/path` - the path part of the service endpoint url. An example value could be `cxfcdi`,
+* `api.kubernetes.io/scheme` - the scheme part of the service endpoint url. Some values could be `http` or `https`.
+* `api.kubernetes.io/prototype` - the protocol of the service. Example values can be `SOAP` or `REST`,
+* `description.api.kubernetes.io/path` - the path part of the service description document’s endpoint. It is a pretty safe assumption that the service self-documents. An example value for a swagger 2.0 document can be `cxfcdi/swagger.json`,
+* `description.api.kubernetes.io/language` - the type of Description Language used. Example values can be `WSDL`, `WADL`, `SwaggerJSON`, `SwaggerYAML`.
+
+The fragment below is taken from the service section of the kubernetes.json were these annotations are used
+
+    ...
+    "objects" : [ {
+      "apiVersion" : "v1",
+      "kind" : "Service",
+      "metadata" : {
+        "annotations" : {
```

Comment:
Didn't convince, sorry.  `api` makes it appear to be about the kube api.

How about, final answer:
```
         "api.service.kubernetes.io/protocol" : "REST",
          "api.service.kubernetes.io/scheme" "http",
          "api.service.kubernetes.io/path" : "cxfcdi",
          "api.service.kubernetes.io/description-path" : "cxfcdi/swagger.json",
          "api.service.kubernetes.io/description-language" : "SwaggerJSON"
```